weather_dl/download_pipeline/manifest_test.py

```diff
@@ -90,3 +97,40 @@ def is_valid_json(self, file: t.IO) -> None:
             json.dumps(json.load(file))
         except json.JSONDecodeError:
             self.fail('JSON is invalid.')
+
+
+# noinspection PyBroadException
+class GCSManifestTest(unittest.TestCase):
+
+    def test_write_valid_json(self):
+
+        # create temporary bucket
+        client = storage.Client()
+        bucket_name = str(uuid.uuid4())
+        path = f"/{str(uuid.uuid4())}/location.json"
+        url = f"gs://{bucket_name}{path}"
+        location = Location(url)
+        tmp_bucket = client.create_bucket(client.bucket(bucket_name))
+        logging.debug(f"Created temporary bucket {bucket_name}")
+
+        try:
+            manifest = GCSManifest(location=location)
+            status = make_download_status(location)
+            manifest._update(status)
+
+            with gcsio.GcsIO().open(url, "rb") as f:
+                j = json.load(f)
+                self.assertEqual(j, status._asdict())
+        finally:
+            try:
+                # clear the bucket's files
+                for blob in tmp_bucket.list_blobs():
+                    blob.delete()
+                # delete the bucket
+                tmp_bucket.delete()
+                logging.debug(f"Deleted temporary bucket {bucket_name}")
+            except Exception as e:
+                logging.error(e)
+                logging.error(f"Error deleting temporary bucket {bucket_name},"
+                              f" to avoid unnecessary Cloud Storage charges "
+                              f"make sure to delete it manually.")
```

Comment:
How about we include this in a test case `tearDown` method?

Reply:
Similarly, the bucket setup could be moved to a `setUp`.

Reply:
You're right. Sorry I wrote these at 2 AM lol, I'll update the CL